src/options/filter.rs

```diff
@@ -3,20 +3,21 @@
 use crate::fs::DotFilter;
 use crate::fs::filter::{FileFilter, SortField, SortCase, IgnorePatterns, GitIgnore};
 
-use crate::options::{flags, OptionsError};
-use crate::options::parser::MatchedFlags;
+use crate::options::OptionsError;
+
+use super::parser::Opts;
 
 
 impl FileFilter {
 
     /// Determines which of all the file filter options to use.
-    pub fn deduce(matches: &MatchedFlags<'_>) -> Result<Self, OptionsError> {
+    pub fn deduce(matches: &Opts, strictness: bool) -> Result<Self, OptionsError> {
```

Comment:
`clap` does have an `env` feature to pull values from the environment if they are not overwritten by the corresponding argument. This also still respects default values set via clap. Seeing all this special treatment of the environment variables, I wonder if we shouldn't use this feature to streamline this a bit and make those things also available to the user via the CLI arguments. Could be something for a followup too, though.

Reply:
Well this needs to be discussed more, Your point is great tho

Reply:
But again probably better in a follow-up PR :)